pkg/controllers/gateway/gateway_controller.go

```diff
@@ -226,6 +252,10 @@ func (r *GatewayReconciler) reconcileDownstreamFromUpstreamGateway(ctx context.C
 	if err != nil {
 		return false, metav1.ConditionFalse, clusters, fmt.Errorf("failed to get managed hosts : %s", err)
 	}
+	if len(managedHosts) == 0 {
+		return false, metav1.ConditionFalse, clusters, fmt.Errorf("no managed hosts found")
+	}
```

Comment:
The "Programmed" status is currently being set to false only when we don't have a valid ManagedZone for any hosts and does not check if DNS is actually setup. This is true on main and this PR.

My understanding based on https://github.com/Kuadrant/multicluster-gateway-controller/pull/251#discussion_r1229393328, and external conversations is that we would expect that to remain false until at least on managed host is found for a listener and a DNSRecord created and ready. Since that's not happening on main, I'm not fixing it in this PR but will add a follow on issue for it.

Reply:
@philbrookes @maleck13 fyi